### PR TITLE
improve unit test for caching

### PR DESCRIPTION
**Pull request by earl-warren**
- Configure `setting.CacheService.TTL` which will force the code to go trough the caching mechanism.
- Remove the TODO and uncomment the test code.

(cherry picked from commit a201f2f1896c1a086f3e37f86b1306d44c07f2e4)

Refs: https://codeberg.org/forgejo/forgejo/pulls/974

### CODE REVIEW

File: modules/cache/cache_test.go [hunk 1]
```diff
@@ -54,26 +55,25 @@ func TestGetString(t *testing.T) {
...
-	// assert.NoError(t, err)
-	// assert.Equal(t, "some data", data)
-
-	// TODO: uncommented code works in IDE but not with go test
+	data, err = GetString("key", func() (string, error) {
+		return "", fmt.Errorf("some error")
+	})
+	assert.NoError(t, err)
+	assert.Equal(t, "some data", data)
+	Remove("key")
```
Comment (KN4CK3R) [inline modules/cache/cache_test.go:76]: This test could fail if somehow this thread idles for 30 nanoseconds (`setting.CacheService.TTL = 30`)?

Comment (delvh) [reply modules/cache/cache_test.go:76]: Isn't that in milliseconds or even seconds?
But yes, theoretically it is possible that this test fails randomly.

Comment (KN4CK3R) [reply modules/cache/cache_test.go:76]: TTL is [`time.Duration`](https://pkg.go.dev/time#Duration)
> A Duration represents the elapsed time between two instants as an int64 nanosecond count.

Comment (KN4CK3R) [reply modules/cache/cache_test.go:76]: These tests cover the "is in the cache and does not need to be computed" case, so the TTL should be irrelevant I think. A simple `setting.CacheService.TTL = 24 * time.Hour` should prevent random failures.

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**